packages/spacecat-shared-data-access/src/models/importer/import-job.js

```diff
@@ -10,161 +10,163 @@
-    if (!isInteger(successCount)) {
-      throw new Error(`Invalid success count during update: ${successCount}`);
+   * Updates the success count of the ImportJob.
+   * @param {number} successCount - The new success count.
+   */
+  importJob.updateSuccessCount = (successCount) => updateState('successCount', successCount, (value) => {
+    if (!isInteger(value)) {
+      throw new Error(`Invalid success count during update: ${value}`);
     }
-
-    self.state.successCount = successCount;
-    self.touch();
-
-    return self;
-  };
+  });
 
   /**
-     * Updates the failed count of the ImportJob.
-     * @param {number} failedCount - The new failed count.
-     * @returns {ImportJob} The updated ImportJob object.
-     */
-  self.updateFailedCount = (failedCount) => {
-    if (!isInteger(failedCount)) {
-      throw new Error(`Invalid failed count during update: ${failedCount}`);
+   * Updates the failed count of the ImportJob.
+   * @param {number} failedCount - The new failed count.
+   * @returns {{ImportJob}} The updated ImportJob object.
+   */
+  importJob.updateFailedCount = (failedCount) => updateState('failedCount', failedCount, (value) => {
+    if (!isInteger(value)) {
+      throw new Error(`Invalid failed count during update: ${value}`);
     }
-
-    self.state.failedCount = failedCount;
-    self.touch();
-
-    return self;
-  };
+  });
 
   /**
-     * Updates the import queue id of the ImportJob.
-     * @param {string} importQueueId - The new import queue id.
-     * @returns {ImportJob} The updated ImportJob object.
-     */
-  self.updateImportQueueId = (importQueueId) => {
-    if (!hasText(importQueueId)) {
-      throw new Error(`Invalid import queue id during update: ${importQueueId}`);
-    }
-
-    self.state.importQueueId = importQueueId;
-    self.touch();
-
-    return self;
-  };
-  return Object.freeze(self);
+   * Updates the import queue id of the ImportJob.
+   * @param {string} importQueueId - The new import queue id.
+   * @returns {ImportJob} The updated ImportJob object.
+   */
+  importJob.updateImportQueueId = (importQueueId) => updateState(
+    'importQueueId',
+    importQueueId,
+    (value) => {
+      if (!hasText(importQueueId)) {
+        throw new Error(`Invalid import queue id during update: ${value}`);
+      }
+    },
+  );
+
+  return Object.freeze(importJob);
 };
 
 /**
  * Creates a new ImportJob object.
- * @param {Object} importJobData - The data for the ImportJob object.
- * @returns {ImportJob} The new ImportJob object.
+ * @param {Object} data - The data for the ImportJob object.
+ * @returns {Readonly<ImportJob>} The new ImportJob object.
+ * @throws {Error} If any of the required fields are invalid.
  */
 export const createImportJob = (data) => {
+  // Define a list of data type validators for each import option
+  const ImportOptionTypeValidator = {
+    [ImportOptions.ENABLE_JAVASCRIPT]: (value) => {
+      if (![true, false, 'true', 'false'].includes(value)) {
```

Comment:
Can `isBoolean` be used instead? Should never be a string.

Reply:
I was wondering about how the values are being initialized when the request comes in.  Is there a jackson layer that is converting the body of the json into the correct types?  What if someone sends in a { enableJavascript: "true" } vs { enableJavascript: true }.

Reply:
That happens in https://github.com/adobe/helix-shared/tree/main/packages/helix-shared-body-data